File: m2-h2/src/main/java/ru/yegorpilipenko/otus/spring/m2h2/document/Author.java
package ru.yegorpilipenko.otus.spring.m2h2.document;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.RequiredArgsConstructor;

import javax.annotation.Nonnull;

import static lombok.AccessLevel.PRIVATE;

@Data
@RequiredArgsConstructor
@NoArgsConstructor(access = PRIVATE, force = true)
public final class Author {

    @Nonnull
    private final String firstName;
    @Nonnull
    private final String lastName;

}

File: m2-h1/src/main/kotlin/ru/yegorpilipenko/otus/spring/m2h1/entity/Author.kt
package ru.yegorpilipenko.otus.spring.m2h1.entity

import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType.SEQUENCE
import javax.persistence.Id
import javax.persistence.SequenceGenerator
import javax.persistence.Table

@Entity
@Table(name = "AUTHORS", schema = "OTUS_LIBRARY")
data class Author(

        @Id
        @Column(name = "ID", unique = true, nullable = false)
        @GeneratedValue(strategy = SEQUENCE, generator = "AUTHORS_ID_GENERATOR")
        @SequenceGenerator(
                name = "AUTHORS_ID_GENERATOR",
                sequenceName = "OTUS_LIBRARY.AUTHORS_ID_SQ",
                allocationSize = 1,
                initialValue = 1
        )
        var id: Long? = null,

        @Column(name = "FIRST_NAME", nullable = false)
        val firstName: String,

        @Column(name = "LAST_NAME", nullable = false)
        val lastName: String

)
